build-js.sh

```diff
@@ -73,876 +72,49 @@ OUT="$PWD/out"
 
 echo "generate embedded intrinsics module"
 
-python3 $BINARYEN_SCRIPTS/embedwast.py $BINARYEN_SRC/passes/wasm-intrinsics.wast $BINARYEN_SRC/passes/WasmIntrinsics.cpp
+python $BINARYEN_SCRIPTS/embedwast.py $BINARYEN_SRC/passes/wasm-intrinsics.wast $BINARYEN_SRC/passes/WasmIntrinsics.cpp
 
 echo "compiling source files"
 
-mkdir -p ${OUT}
+# internal source files or directories to exclude
+EXCLUDE=(
+  "$BINARYEN_SRC/support/archive.cpp"
+  "$BINARYEN_SRC/support/command-line.cpp"
+  "$BINARYEN_SRC/support/path.cpp"
+  "$BINARYEN_SRC/tools/"
+)
+
+# external source files to include
+INCLUDE=()
+
+mapfile -t SOURCES < <(ls $BINARYEN_SRC/**/*.cpp)
+for i in ${!SOURCES[@]}; do
```

Comment:
I'm not huge find of bash script once they start to use these kinds of advanced features... but I think we can make an exception.

I wonder if we can't just use cmake here instead.. but that is a separate issue.

Reply:
On the binaryen.js bot I've been using [this node.js script](https://github.com/AssemblyScript/binaryen.js/blob/master/scripts/build.js) to do about the same, but recently switched back to utilize build-js.sh there, in turn leading to this PR. If there's interest, I can try to port the JS script over to the binaryen repo instead, and maybe add a `package.json` so one can run `npm run build`?

Reply:
I'd rather just make the cmake build work.  I don't see why emscripten should be able to build binaryen in the normal way.. using cmake + ninja.

Reply:
I'm not familiar with cmake and ninja unfortunately. So far I've appreciated that I can run `bash build-js.sh` and it "just works" even on Windows or inside of a docker container that might not have the full set of build tools.